extras/kdf-keys/pbkdf2.c

```diff
@@ -111,19 +111,24 @@ int main(int argc, char *argv[])
 	/* Read password char by char.
 	 *
 	 * Doing in this way we make sure that blanks (even null bytes) end up
-	 * in the password
+	 * in the password.
+	 *
+	 * passwords containing just a bunch of spaces are valid
 	 */
-	int j = 0;
 	while (j < (BUFFER_SIZE + 1)) {
 		char c = getchar();
 		if (c == EOF) break;
 		pass[j] = c;
 		j++;
 	}
-	if (j == BUFFER_SIZE + 1) {
+	if (j >= BUFFER_SIZE + 1) {
```

Comment:
Shouldn't the check be on BUFFER_SIZE? 
Should the password have 1024 chars, it will loose one char, when it is made sure that the string array ends with `\0`

Reply:
Actually the max password is BUFFER_SIZE - 1. 
Anyway I will double check it.

Reply:
I think the logic works, i added a test checking that a password of 1023 characters is accepted and that 1024 characters is refused.

Anyway i removed the `+ 1` fixing the equality operators.

Reply:
Ah, right. Didn't take the while loop into account